app/models/status.rb

```diff
@@ -1,3 +1,14 @@
 class Status < ApplicationRecord
   validates :body, length: { maximum: 300 }, presence: true
+  has_many :replies, class_name: 'Status', foreign_key: 'replied_to_status_id', dependent: :destroy
+  belongs_to :replied_to_status, class_name: 'Status', optional: true
+
+  belongs_to :user
+  has_many :media, dependent: :destroy
+  accepts_nested_attributes_for :media, allow_destroy: true
```

Comment:
```suggestion
  accepts_nested_attributes_for :media, allow_destroy: true, reject_if: proc { |attributes| attributes['url'].blank? }
```

You might want to add "reject_if " to avoid saving empty media to the database.